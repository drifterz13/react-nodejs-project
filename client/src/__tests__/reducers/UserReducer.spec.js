import userReducer from '../../store/reducers/user'

describe('User reducer', () => {
  describe('DEFAULT', () => {
    test('must return initial state', () => {
      const action = { type: 'none_action' }
      const initialState = { info: {}, authenticated: false }
      expect(userReducer(undefined, action)).toMatchSnapshot()
    })
  })
  describe('LOGGED_IN', () => {
    test('return the correct state', () => {
      const payload = {
        username: 'test',
        email: 'test@test.com',
        password: 'test'
      }
      const action = { type: 'LOGGED_IN', payload }
      expect(userReducer(undefined, action)).toMatchSnapshot()
    })
  })
  describe('SET_PREF', () => {
    test('return the update preferences', () => {
      const payload = {
        language: 'EN',
        timezone: '-12.0',
        currency: 'USD',
        profileType: 'all',
        messageType: 'follow',
        categoryType: 'enable'
      }
      const action = { type: 'SET_PREF', payload }
      expect(userReducer(undefined, action)).toMatchSnapshot()
    })
  })
})