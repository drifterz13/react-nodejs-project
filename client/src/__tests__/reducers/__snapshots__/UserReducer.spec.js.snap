// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User reducer DEFAULT must return initial state 1`] = `
Object {
  "authenticated": false,
  "info": Object {},
}
`;

exports[`User reducer LOGGED_IN return the correct state 1`] = `
Object {
  "authenticated": true,
  "info": Object {
    "email": "test@test.com",
    "password": "test",
    "username": "test",
  },
}
`;

exports[`User reducer SET_PREF return the update preferences 1`] = `
Object {
  "authenticated": false,
  "info": Object {
    "preferences": Object {
      "categoryType": "enable",
      "currency": "USD",
      "language": "EN",
      "messageType": "follow",
      "profileType": "all",
      "timezone": "-12.0",
    },
  },
}
`;
